projects/aws/upgrader/upgrade/kubeadm.go

```diff
@@ -293,3 +377,45 @@ func (u *InPlaceUpgrader) restoreCoreDNSConfig(ctx context.Context, cmpDir strin
 	logger.Info("restored coreDNS config successfully!")
 	return nil
 }
+
+func newTrue() *bool {
+	t := true
+	return &t
+}
+
+func generateKubeAdmConfCM(clusterConfig string) corev1.ConfigMap {
+	return corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kubeadmCMName,
+			Namespace: kubeSystemNS,
+		},
+		Data: map[string]string{"ClusterConfiguration": clusterConfig},
+	}
+}
+
+func generateKubeAdmUpgradeConfig(version string) UpgradeConfiguration {
+	preflightErrorsList := []string{"CoreDNSUnsupportedPlugins", "CoreDNSMigration"}
+	return UpgradeConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       upgradeConfigurationKind,
+			APIVersion: kubeadmAPIv1beta4,
+		},
+		Apply: UpgradeApplyConfiguration{
+			KubernetesVersion:         version,
+			AllowExperimentalUpgrades: newTrue(),
```

Comment:
nit: can you use `ptr.Bool(true)` instead?
```suggestion
			AllowExperimentalUpgrades: ptr.Bool(true),
```

Reply:
um don't think that's standard lib function. You mean use this: "github.com/aws/eks-anywhere/pkg/utils/ptr" .?

Reply:
you can use something like this https://go.dev/play/p/PiBOfDuYBRs